src/stridelayout.jl

```diff
@@ -220,7 +220,7 @@ function stride_rank(::Type{T}) where {T}
         return stride_rank(parent_type(T))
     end
 end
-stride_rank(::Type{Array{T,N}}) where {T,N} = nstatic(Val(N))
+stride_rank(::Type{DenseArray{T,N}}) where {T,N} = nstatic(Val(N))
```

Comment:
This needs to be subtyping `DenseArray`

`stride_rank(::Type{<:DenseArray{T,N}}) where {T,N} = nstatic(Val(N))`

Reply:
Ah, that makes sense